src/libs/OptionsListUtils.ts

```diff
@@ -733,67 +810,57 @@ function sortCategories(categories) {
      * A recursive function to convert hierarchy into an array of category objects.
      * The category object contains base 2 properties: "name" and "enabled".
      * It iterates each key one by one. When a category has subcategories, goes deeper into them. Also, sorts subcategories alphabetically.
-     *
-     * @param {Object} initialHierarchy
-     * @returns {Array<Object>}
      */
-    const flatHierarchy = (initialHierarchy) =>
-        _.reduce(
-            initialHierarchy,
-            (acc, category) => {
-                const {name, ...subcategories} = category;
-
-                if (!_.isEmpty(name)) {
-                    const categoryObject = {
-                        name,
-                        enabled: lodashGet(categories, [name, 'enabled'], false),
-                    };
-
-                    acc.push(categoryObject);
-                }
+    const flatHierarchy = (initialHierarchy: Hierarchy) =>
+        Object.values(initialHierarchy).reduce((acc: Category[], category) => {
+            const {name, ...subcategories} = category;
+            if (name) {
+                const categoryObject: Category = {
+                    name,
+                    enabled: categories[name].enabled ?? false,
```

Comment:
Category should have been accessed safely, rare case where `categories[name]` is undefined. ref: https://github.com/Expensify/App/issues/36622